m4/curl-compilers.m4

```diff
@@ -158,7 +158,7 @@ AC_DEFUN([CURL_CHECK_COMPILER_GNU_C], [
     flags_dbg_all="$flags_dbg_all -gvms"
     flags_dbg_yes="-g"
     flags_dbg_off=""
-    flags_opt_all="-O -O0 -O1 -O2 -O3 -Os"
+    flags_opt_all="-O -O0 -O1 -O2 -O3 -Os -Og"
```

Comment:
Thanks, how could I overlook -Ofast when studying the GCC optimization flags for almost an hour yesterday...
Changes pushed.